profiler/src/ProfilerEngine/Datadog.Profiler.Native/FrameStore.cpp

```diff
@@ -932,3 +1125,417 @@ std::pair<std::string, std::string> FrameStore::GetManagedTypeName(ICorProfilerI
+
+        case ELEMENT_TYPE_I:
+            StrAppend(buffer, "nint", cchBuffer);
+            break;
+
+        case ELEMENT_TYPE_OBJECT:
+            StrAppend(buffer, "object", cchBuffer);
+            break;
+
+        case ELEMENT_TYPE_STRING:
+            StrAppend(buffer, "string", cchBuffer);
+            break;
+
+        case ELEMENT_TYPE_TYPEDBYREF:
+            StrAppend(buffer, "refany", cchBuffer);
+            break;
+
+        case ELEMENT_TYPE_CLASS:
+        case ELEMENT_TYPE_VALUETYPE:
+        case ELEMENT_TYPE_CMOD_REQD:
+        case ELEMENT_TYPE_CMOD_OPT:
+        {
+            mdToken token;
+            char classname[260];
+
+            classname[0] = '\0';
+            signature += CorSigUncompressToken(signature, &token);
+            if (typeToken != NULL)
+            {
+                *typeToken = token;
+            }
+
+            HRESULT hr;
+            WCHAR zName[260];
+            if (TypeFromToken(token) == mdtTypeRef)
+            {
+                mdToken resScope;
+                ULONG length;
+                hr = pMDImport->GetTypeRefProps(token,
+                                                &resScope,
+                                                zName,
+                                                260,
+                                                &length);
+            }
+            else
+            {
+                hr = pMDImport->GetTypeDefProps(token,
+                                                zName,
+                                                260,
+                                                NULL,
+                                                NULL,
+                                                NULL);
+            }
+            if (SUCCEEDED(hr))
+            {
+                StrAppend(buffer, shared::ToString(shared::WSTRING(zName)).c_str(), cchBuffer);
+            }
+            else
+            {
+                StrAppend(buffer, "?", cchBuffer);
+            }
+        }
+        break;
+
+        case ELEMENT_TYPE_SZARRAY:
+            signature = ParseElementType(pMDImport, signature, classTypeArgs, methodTypeArgs, &eType, buffer, cchBuffer, typeToken);
+            StrAppend(buffer, "[]", cchBuffer);
+            break;
+
+        case ELEMENT_TYPE_ARRAY:
+        {
+            ULONG rank;
+            signature = ParseElementType(pMDImport, signature, classTypeArgs, methodTypeArgs, &eType, buffer, cchBuffer, typeToken);
+            rank = CorSigUncompressData(signature);
+
+            // The second condition is to guard against overflow bugs & shut up PREFAST
+            if (rank == 0 || rank >= 65536)
+            {
+                StrAppend(buffer, "[?]", cchBuffer);
+            }
+            else
+            {
+                ULONG* lower;
+                ULONG* sizes;
+                ULONG numsizes;
+                ULONG arraysize = (sizeof(ULONG) * 2 * rank);
+
+                lower = (ULONG*)_alloca(arraysize);
```

Comment:
not sure `_alloca` exists on linux., if I recall it `alloca` instead

Reply:
It seems to exist  :^)